### DISCUSSION TITLE
Face swap generic help

### DISCUSSION THREAD

**Discussion by JoelHJ** [lllyasviel/Fooocus#1097] [General]
First.. - wow what a app. I just found you guys and have downloaded and installed Fooocus and its amazing. I have been using Midjourney for so long but found this. My biggest gripe with all AI images is the face swap using the same face on new images. My dream is to be able to use a face and swap it but keep 90% ish of the actual face so one can see its the same face. This is something I have yet to master. I get some strange results and keep changing the setting but just can't get AI to swap a face close to original. I even swap face.. than use the new generated face as the new input image and by rinse and repeat hope that it would get closer and close to original face but no. Any ideas if anyone has time would be amazing.. any ideas at all or maybe we are just not there yet. 

Only thing I have seen on youtube is something called Conditioning Mask Strength" or "Unmasked Conditioning that this guy uses for faces here https://www.youtube.com/watch?v=VdDhGjkKD1A but don't understand how on earth i could get that to work

Thank you so much for your time and amazing work

**Comment by elgroucho** [comment]
Try this in Fooocus https://www.youtube.com/watch?v=wOBcNPFuB70
That video is in Chinese, but if you click on the icons on the same location as on the video it will be obvious. 

Otherwise, I know Automatic1111 with the ReActor extension works very well. You can import the images from Fooocus into Automatic1111 and do the face swap there if Fooocus' face swap doesn't give you good results.

**Reply by elgroucho** [reply]
This other video is better. The UI matches the latest version of Fooocus. Though I still get better results with Automatic1111 and ReActor for face swaps. 
https://youtu.be/ky5ZB-mqZKM?si=Gy5XoWY7uteN7T-p

**Reply by KitsonBroadhurst** [reply]
An english version of the first one linked:  https://www.youtube.com/watch?v=nWvOzBG1k_k

**Reply by JoelHJ** [reply]
Thank you so much guys

**Reply by Amit30swgoh** [reply]
Why did she dragged the generated black women into upscale and also uploaded her image to inpaint or outpaint.

**Comment by zjtdev** [comment]
You can also make use of this online face swap tool, as it delivers excellent results:
https://remaker.ai/face-swap-free

**Comment by justnpT** [comment]
I agree, this is difficult even Today. Fooocus face swap works alrightish with frontal face images, but it gets trickier when the person is looking left or right or up or upright or upleft. Then the face swap doesn't happen naturally, it tends to change the original image.

https://www.youtube.com/watch?v=nWvOzBG1k_k in this video there is some helpful description but getting results I want is something I'm still struggling to accomplish

**Comment by infinity0** [comment]
Try what I described in #2644. I figured out that Upscale works much better than Vary (Subtle) which is what the video in Chinese above suggests.

**Comment by GOODDAYDAY** [comment]
Fooocus is indeed very good, but it requires high technical and computer specifications. If you don't have a GPU, your computer may even crash. 

For beginners, I would recommend Deepswap instead, an online website that uses a private face swapping model. The deepfake speed and quality are superior to most websites that use open-source models: https://www.deepswap.ai/

**Comment by kalle07** [comment]
i write an articel to use adetailer and reactor
https://civitai.com/articles/5002/face-swap-with-adetailer-and-reactor
